components/calendar/__tests__/index.test.js

```diff
@@ -14,4 +14,17 @@ describe('Calendar', () => {
     const value = onSelect.mock.calls[0][0];
     expect(Moment.isMoment(value)).toBe(true);
   });
+  it('only Valid range should be selectable', () => {
+    const onSelect = jest.fn();
+    const validRange = () => ({
+      startDate: Moment('2018-02-02'),
+      endDate: Moment('2018-02-18'),
+    });
```

Comment:
Should the return of `validRange` is a array?

Reply:
@yesmeck its an object with keys as startDate and endDate
example:
function validRange(current) {
  return {startDate: "moment date", endDate: "moment date"}
}
Do you want me to make it an array instead ?

Reply:
Yes I think it should be a same type as `RangePicker`'s value. Another question is why it is a function?

Reply:
so that you can control valid range dynamically in a way similar to disabledDate

Reply:
will change it to an array

Reply:
I think I should make its api similar to rangepicker with both function and array type prop

Reply:
@yesmeck updated it to be similar to rangepicker's value